Sources/WalletConnectAuth/Types/Session/WCSession.swift

```diff
@@ -62,47 +62,46 @@ struct WCSession: ExpirableSequence {
         return controller.publicKey == peerParticipant.publicKey
     }
     
-    // FIXME
     func hasNamespace(for chain: Blockchain) -> Bool {
-        // TODO
-//        namespaces.contains{$0.chains.contains(chain)}
-        return true
+        return namespaces[chain.namespace] != nil
     }
     
-    // TODO: Remove optional for chain param, it's required now / protocol change
-    func hasNamespace(for chain: Blockchain?, method: String) -> Bool {
-        // TODO
-//        if let chain = chain {
-//            let namespacesIncludingChain = namespaces.filter{$0.chains.contains(chain)}
-//            let methods = namespacesIncludingChain.flatMap{$0.methods}
-//            return methods.contains(method)
-//        } else {
-//            return namespaces
-//                .filter { $0.chains.isEmpty }
-//                .flatMap { $0.methods }
-//                .contains(method)
-//        }
-        return true
+    func hasPermission(forMethod method: String, onChain chain: Blockchain) -> Bool {
+        if let namespace = namespaces[chain.namespace] {
```

Comment:
Could we reused that code with (line 88) somehow? Look's lite pretty identical. For example we could create private extension on `Set<String>`

Reply:
I thought that it may be possible with KeyPaths, by making a single function that receives the KeyPath (`\.methods` or `\.events`) and search using it. But I decided to keep it as it is until we're sure that there will be no more changes in here for now.

We can definitely improve it later.